src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -543,7 +543,7 @@ partition_downloader::find_recovery_material(const remote_manifest_path& key) {
     if (result != download_result::success) {
         throw missing_partition_exception(key);
     }
-    auto orig_rev = recovery_mat.topic_manifest.get_revision();
+    auto orig_rev = recovery_mat.topic_manifest.get_prev_revision();
```

Comment:
Do we ever need to clear `prev_revision`? Trying to understand what happens when partitions finish their recovery, and then something happens and we need to recover again.